GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1311,7 +1326,23 @@ private void ApplyPatchToolStripMenuItemClick(object sender, EventArgs e)
 
         private void GitBashToolStripMenuItemClick1(object sender, EventArgs e)
         {
-            Module.RunBash();
+            if (!GitBash.DropDownButtonPressed)
+            {
+                Module.RunBash();
+            }
+        }
+
+        private void RunShell(object sender, EventArgs e)
+        {
+            try
+            {
+                string shellPath = ShellHelper.GetShellPath(((ToolStripMenuItem)sender).Text);
+                new Executable(shellPath, Module.WorkingDir).Start(createWindow: true);
+            }
+            catch (Exception exception)
+            {
+                MessageBox.Show(_runShellFailed.Text + "\n" + exception.Message,  Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
```suggestion
                MessageBox.Show($"{_runShellFailed.Text}{Environment.NewLine}{exception.Message}", Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
```
And should this `MessageBox` be moved to `GitUI\MessageBoxes.cs`?

Reply:
> And should this `MessageBox` be moved to `GitUI\MessageBoxes.cs`?

👍